uikit/viewmodels/src/main/java/com/sygic/maps/uikit/viewmodels/common/extensions/SdkExtensions.kt

```diff
@@ -173,4 +185,101 @@ fun Application.computePrimaryRoute(routePlan: RoutePlan, routeComputeCallback:
             override fun onPrimaryComputeFinished(router: Router, route: RouteInfo) = routeComputeCallback.invoke(route)
         })
     }
+}
+
+fun List<NaviSignInfo.SignElement>.concat(): String = StringBuilder().apply {
+    this@concat.forEach {
+        if (isNotEmpty()) append(", ")
+        append(it.text)
+    }
+}.toString()
+
+fun List<NaviSignInfo>.getNaviSignInfoOnRoute(): NaviSignInfo? {
+    this.filter { it.isOnRoute }.let { isOnRouteList ->
+        if (isOnRouteList.isEmpty()) {
+            return null
+        }
+
+        return isOnRouteList.firstOrNull { it.backgroundColor != 0 }?.let { it } ?: isOnRouteList[0]
+    }
+}
+
+fun NaviSignInfo.roadSigns(maxRoadSignsCount: Int = 3): List<RoadSignData> {
+    val roadSigns = mutableListOf<RoadSignData>()
+    val hasPic = signElements.find { it.elementType == SignElementType.Pictogram } != null
+    signElements
+        .asSequence()
+        .filter { it.elementType == SignElementType.RouteNumber }
+        .take(if (hasPic) maxRoadSignsCount - 1 else maxRoadSignsCount)
+        .toList()
+        .forEach { signElement ->
+            with(signElement.routeNumberFormat) {
+                if (insideNumber.isNotEmpty()) {
+                    roadSigns.add(
+                        RoadSignData(
+                            roadSignBackgroundDrawableRes(),
+                            insideNumber,
+                            roadSignForegroundColorRes()
+                        )
+                    )
+                }
+            }
+        }
+
+    return roadSigns
+}
+
+fun NaviSignInfo.createInstructionText(): TextHolder {
+    val acceptedSignElements = signElements
+        .filter { element -> element.elementType.let { it == SignElementType.ExitNumber || it == SignElementType.PlaceName || it == SignElementType.OtherDestination } }
+        .sortedByDescending { it.elementType == SignElementType.ExitNumber }
```

Comment:
toto je zvlastne sortovanie, to ma sposobit, ze exity budu "nejako" na zaciatku a ostatne za nimi? Lebo sortovat boolean je asi len to, ze `true` je pred `false` ci?

Reply:
ne nějako, budou :) toto jsme řešili s Maťou Brečkou asi hoďku, nejprve jsme to řešili custom comparatorem, ale nakonec jsme zjistili, že toto je vlastně comparator, takže pro něj je duléžité zda to bude true (1) false (0) nebo -1 ta v tomto případě nenastane, ale to nevadí, protože podle dokumentace je to **stable** sort, takže upředností exit number a ostatní nechá v původním pořadí.

Reply:
no ok vdaka stable sortovaniu to tak bude, ale je to na pohlad ohack sortovanie :)

Reply:
ohack? takto je to lepší?
```kotlin
.sortedWith(Comparator<NaviSignInfo.SignElement>{ a, b ->
                when {
                    a.elementType == NaviSignInfo.SignElement.SignElementType.ExitNumber -> -1
                    else -> 0
                }
            })
```

Reply:
nie, moc nie :) "Zmysluplny" sort by bol `sortedByDescending { it.elementType }`, ale chapem, ze ty ich v skutocnosti triedit nechces, ty chces len zobrazit Exity pred ostatnymi, takze kludne nechaj to povodne. Ten stable sort to vyriesi, aj ked to clovek na prvy pohlad asi nespozna...

Reply:
mohu tam dát alespoň komentář.